tests/fixture/subnet/xsvm.go

```diff
@@ -0,0 +1,53 @@
+// Copyright (C) 2019-2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package subnet
+
+import (
+	"math"
+	"time"
+
+	"github.com/ava-labs/avalanchego/tests/fixture/tmpnet"
+	"github.com/ava-labs/avalanchego/utils/crypto/secp256k1"
+	"github.com/ava-labs/avalanchego/vms/example/xsvm"
+	"github.com/ava-labs/avalanchego/vms/example/xsvm/genesis"
+)
+
+func NewXSVMOrPanic(name string, key *secp256k1.PrivateKey, nodes ...*tmpnet.Node) *tmpnet.Subnet {
+	if len(nodes) == 0 {
+		panic("a subnet must be validated by at least one node")
+	}
+
+	if key == nil {
+		var err error
+		key, err = secp256k1.NewPrivateKey()
+		if err != nil {
+			panic(err)
+		}
+	}
+
```

Comment:
I feel like it looks weird to pass in `nil` in the caller function and override it with a generated key. Why don't we just force the caller to generate the key instead?